src/ol/source/Google.js

```diff
@@ -238,10 +252,48 @@ class Google extends TileImage {
     const timeout = Math.max(expiry - Date.now() - 60 * 1000, 1);
     this.sessionRefreshId_ = setTimeout(() => this.createSession_(), timeout);
 
+    this.setAttributions(this.fetchAttributions.bind(this));
     // even if the state is already ready, we want the change event
     this.setState('ready');
   }
 
+  async fetchAttributions(frameState) {
+    if (
+      frameState.viewHints[ViewHint.ANIMATING] ||
+      frameState.viewHints[ViewHint.INTERACTING]
+    ) {
+      return this.previousViewportAttribution_;
+    }
```

Comment:
I think if you also keep checking for `frameState.animating` here, you won't get duplicate calls for the same extent. 

If that does not work, you could bail out here already when the extent has not changed:
```suggestion
    if (
      frameState.viewHints[ViewHint.ANIMATING] ||
      frameState.viewHints[ViewHint.INTERACTING] ||
      equalsExtent(frameState.extent, this.previousViewportExtent_)
    ) {
      return this.previousViewportAttribution_;
    }
    this.previousViewportExtent_ = frameState.extent.slice(0);
```